server/src/main/java/uk/ac/cam/cl/echo/extrusionfinder/server/preprocessor/ColorSpaceConversion.java

```diff
@@ -0,0 +1,180 @@
+package uk.ac.cam.echo.extrusionfinder.preprocessor;
+
+import java.lang.Math;
+
+import uk.ac.cam.cl.echo.extrusionfinder.server.imagedata.LabImageData;
+import uk.ac.cam.cl.echo.extrusionfinder.server.imagedata.RGBImageData;
+import uk.ac.cam.cl.echo.extrusionfinder.server.imagedata.XYZImageData;
+
+/**
+ * Collection of static colorspace conversion methods.
+ * Includes pairwise conversions between RGB, XYZ and L*a*b*.
+ */
+public class ColorSpaceConversion {
+
+    // The CIE XYZ tristimulus values of the reference white point for D6500
+    // wikipedia.org/wiki/Illuminant_D65#Definition
```

Comment:
How much would it take to give at least a vague description of what all of the fields and methods do, without simply linking to Wikipedia? This code is basically mysterious.

Many of the names used in this code are quite cryptic. Are they even conventional?